controller/app.go

```diff
@@ -55,6 +56,12 @@ func (r *AppRepo) Add(data interface{}) error {
 	}
 	meta := metaToHstore(app.Meta)
 	if err := r.db.QueryRow("INSERT INTO apps (app_id, name, meta, strategy) VALUES ($1, $2, $3, $4) RETURNING created_at, updated_at", app.ID, app.Name, meta, app.Strategy).Scan(&app.CreatedAt, &app.UpdatedAt); err != nil {
+		if e, ok := err.(*pq.Error); ok && e.Code.Name() == "unique_violation" && e.Constraint == "apps_name_idx" {
+			return httphelper.JSONError{
+				Code:    httphelper.ObjectExistsError,
+				Message: fmt.Sprintf("Application with name '%s' already exists.", app.Name),
+			}
```

Comment:
I have just merged #1209 so you should rebase this and change to something like:

```go
return httphelper.ObjectExistsErr(fmt.Sprintf("application %q already exists", app.Name))
```